ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/MigrationsDirectoryUtil.java

```diff
@@ -46,6 +48,32 @@
   private MigrationsDirectoryUtil() {
   }
 
+  /**
+   * Returns the migrations directory path based on the provided config and
+   * config file path. This method does not perform validation on the provided
+   * directory location; that responsibility is left to downstream methods.
+   */
+  public static String getMigrationsDir(
+      final String configFilePath,
+      final MigrationConfig config
+  ) {
+    final String migrationsDir = config.getString(MigrationConfig.KSQL_MIGRATIONS_DIRECTORY_PATH);
+    if (migrationsDir != null && !migrationsDir.isEmpty()) {
+      return migrationsDir;
+    } else {
+      return getMigrationsDirFromConfigFile(configFilePath);
+    }
+  }
+
+  /**
+   * Returns the migrations directory path assuming the default file structure
+   * (as created by the 'ksql-migrations new-project' command). This method should
+   * NOT be called directly outside of testing; prefer
+   * {@link MigrationsDirectoryUtil#getMigrationsDir(String, MigrationConfig)}
+   * instead as users can override the default location returned by this method
+   * in their migrations config if they wish.
+   */
+  @VisibleForTesting
```

Comment:
Would it make sense to mark this function as deprecated?  So that it could be switched to package protected or something?

Reply:
It's not deprecated. Ideally it'd be a private helper method in this file, but it's needed from MigrationsTest.java, thus the annotation.